### PR TITLE
Rose suite run proposals cylc flow.rc

### PR DESCRIPTION
**Pull request by wxtim**
Follow on from #40.
- [x] Create a specification document for the new `cylc-flow.rc` file.

### CODE REVIEW

File: docs/proposal-rose-suite-run.md [hunk 1]
```diff
@@ -136,16 +229,28 @@ sources on installation. Other things to consider:
...
-* Restart a suite at a given checkpoint.
-* Warm start a suite.
-* Cold start a suite.
-* Clean start a suite.
+a single command (e.g. `cylc flow`, `cylc go`?) to with
+appropriate command switches should be agreed on in a pull
+request to this repository modifying [this document](rose-suite-run-proposal/future-cli-conventions.md)
+
+## Replace Rose Fileinstall & rose-suite.conf
+Create a new file to provide jinja2 data to `cylc-flow.rc` - replaces
```
Comment (matthewrmshin) [inline docs/proposal-rose-suite-run.md]:
```suggestion
Create a new mechanism to provide jinja2 data to `cylc-flow.rc` - replaces
```

File: docs/rose-suite-run-proposal/cylc-flow-rc.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+```
+
+### Top level sections to copy from `global.rc` to `cylc-flow.rc`
+
+It is likely that most users will continue to have these set by site admins.
+
+```ini
+[task events]
+  ...
+[test battery]
```
Comment (matthewrmshin) [inline docs/rose-suite-run-proposal/cylc-flow-rc.md]: Can remove? Only relevant for development.

File: docs/rose-suite-run-proposal/cylc-flow-rc.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+  ...
+[suite host self-identification]
```
Comment (matthewrmshin) [inline docs/rose-suite-run-proposal/cylc-flow-rc.md]: Move to a new `[suite platform]` section?

File: docs/rose-suite-run-proposal/cylc-flow-rc.md [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+  ...
+[authentication]
```
Comment (matthewrmshin) [inline docs/rose-suite-run-proposal/cylc-flow-rc.md]: Should become `[authorisation]`.

File: docs/rose-suite-run-proposal/cylc-flow-rc.md [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+[vizualization]
+  ...
+```
+
+### Top level sections to copy from `global.rc` to `cylc-flow.rc`
+
+It is likely that most users will continue to have these set by site admins.
+
+```ini
+[task events]
```
Comment (matthewrmshin) [inline docs/rose-suite-run-proposal/cylc-flow-rc.md]: Can this become the default `[runtime][[root]][[[events]]]` settings?

File: docs/rose-suite-run-proposal/cylc-flow-rc.md [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+for the purpose of keeping the name "platforms" conistent. This is expected to
+be set only by system administrators.
+
+```ini
+[suite platforms]
+```
+
+
+### Top level sections to merge in a more complex way
+#### `[cylc]`
```
Comment (matthewrmshin) [inline docs/rose-suite-run-proposal/cylc-flow-rc.md]: `[cylc]` should probably become `[general]` or something like that.

Comment (hjoliver) [reply docs/rose-suite-run-proposal/cylc-flow-rc.md]: Yeah, `[cylc]` as a config section heading has annoyed the hell out of me for ages. (My fault too, back in time...)

File: docs/proposal-rose-suite-run.md [hunk 7]
```diff
@@ -24,27 +22,75 @@ The logical step is therefore to move these extra functionalities provided by
...
+  - [ ] Summarize debate on ticket.
+  - [ ] Create a PR against cylc-admin creating a specification for `cylc flow`
+
+
+- [ ] Create a `cylc flow` CLI.
+  - [ ] Include suite validation in `cylc flow` CLI.
+
+
+- [ ] Replace `rose-suite.conf`.
+  - [ ] Create an alternative jinja2 format and tests for this format.
```
[the remark this answers is not in the document]
Comment (hjoliver) [reply docs/proposal-rose-suite-run.md]: May be able go straight to an importable Python module?

File: docs/proposal-rose-suite-run.md [hunk 8]
```diff
@@ -24,27 +22,75 @@ The logical step is therefore to move these extra functionalities provided by
...
+  - [ ] Create a rose-suite.conf parsing plugin for cylc flow.
+
+
 ## Functionalities to Consider
 
 These functionalities are currently provided by Rose, but should really be part
 of Cylc:
-* On start up and reload, install suite on suite cluster.
+* On start up and reload, install suite on suite server cluster (cylc servers).
```
[the remark this answers is not in the document]
Comment (hjoliver) [reply docs/proposal-rose-suite-run.md]: I'd prefer "suite run hosts" or "suite run platform". 

I'm still not sure about "platform" in general.  I forget the original argument agains "cluster" ... but maybe "cluster" is OK even for a single host (the minimum size limit of a cluster!)

File: docs/rose-suite-run-proposal/cylc-flow-rc.md (hunk 3 above)
Comment (hjoliver) [reply docs/rose-suite-run-proposal/cylc-flow-rc.md]: See above ... but maybe "platform" is OK ... undecided :thinking:

File: docs/rose-suite-run-proposal/cylc-flow-rc.md (hunk 6 above)
Comment (wxtim) [reply docs/rose-suite-run-proposal/cylc-flow-rc.md]: What about `[cylc system/general/settings]`? Do we want to break back compatibility this radically?

File: docs/proposal-rose-suite-run.md [hunk 9]
```diff
@@ -24,27 +22,75 @@ The logical step is therefore to move these extra functionalities provided by
...
+
+- [ ] Implement new `cylc-flow.rc` form. [Cylc-flow #3260](https://github.com/cylc/cylc-flow/issues/3260)
+  - [ ] Check old tests for `global.rc` & `suite.rc` to ensure that functionality
+    is not lost.
+  - [ ] Devise tests for the new flow.rc
+  - [ ] Create new config parser, perhaps called `cylc-flow-rc-parser.py`
+
+
+- [ ] Implement Cluster support functionality. [Cylc-flow #2199](https://github.com/cylc/cylc-flow/issues/2199)
+  - [ ] Modify `task_job_mgr.py` to use the new variables.
```
[the remark this answers is not in the document]
Comment (wxtim) [reply docs/proposal-rose-suite-run.md]: and friends? Would you care to use your domain knowledge to help me a bit more?
Are these the files in the result of `grep -lrinP "cylc.flow.cfgspec"`

File: docs/proposal-rose-suite-run.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (wxtim) [reply docs/proposal-rose-suite-run.md]: The idea is that platforms is a more general/neutral term, and that, although clusters can include clusters of 1, that generally the term implies clusters of size >> 1.
I think that a customer hearing cluster will automatically assume we are talking about HPC/Large Compute resource, whereas "platfoms" covers submitting jobs to the raspi on my desk, my smartwatch, my desktop, Matt's desktop, a cylc server, **or** an HPC or SPICE system.

[the remark this answers is not in the document]
Comment (wxtim) [reply docs/proposal-rose-suite-run.md]: I've made it "suite run platform" for the time being.

File: docs/rose-suite-run-proposal/cylc-flow-rc.md (hunk 5 above)
Comment (wxtim) [reply docs/rose-suite-run-proposal/cylc-flow-rc.md]: Why not?

File: docs/rose-suite-run-proposal/cylc-flow-rc.md (hunk 4 above)
Comment (wxtim) [reply docs/rose-suite-run-proposal/cylc-flow-rc.md]: Agreed, with spelling "authorization" which is acceptable (if archaic) in all forms of English.

Comment (sadielbartholomew) [reply docs/rose-suite-run-proposal/cylc-flow-rc.md]: That's a truly beautiful depiction of a can of worms, @wxtim :laughing: Thank you for your commitment to humour!

File: docs/rose-suite-run-proposal/cylc-flow-rc.md [hunk 10]
```diff
@@ -0,0 +1,191 @@
...
+
+## Preamble
+
+To reduce changes required for end users this file format is based on
+`suite.rc`: Where aspects of the file are described as unchanged this implies
+that file contents are exactly the same as they would be in a `suite.rc`.
+Users of the `global.rc` are usually admins or power users and thus changes
+to the format of this file are preferable.
+
+It should be noted that although it will be possible to modify all settings in
```
[the remark this answers is not in the document]
Comment (wxtim) [reply docs/rose-suite-run-proposal/cylc-flow-rc.md]: I can totally imagine this being popular, and as someone who may end up deploying the site.rc I think there may be some merit to it. It shouldn't be too difficult to do, either.

File: docs/proposal-rose-suite-run.md [hunk 11]
```diff
@@ -136,16 +231,28 @@ sources on installation. Other things to consider:
...
 Automatic suite validation should become the default behaviour for `cylc run`
 and `cylc reload`. Things to consider:
-* Is the *strict* mode necessary by default?
-* `rose suite-run` does not currently validate Rose items. This should be
-  changed to ensure that all aspects of the suite are validated uniformly.
+* `rose suite-run` does not currently validate Rose apps against their
+  metadata. Rose apps should be validated against their metadata by default,
+  but contain switches in the manner of bandit or pylint in a config file.
+  (perhaps `cylc-validate.yml` or similar)
+* Creators of largers suites may want to turn this because the validation
```
Comment (kinow) [inline docs/proposal-rose-suite-run.md]: may want to turn this on or off?

Comment (wxtim) [reply docs/proposal-rose-suite-run.md]: I've actuall removed this after a conversation with Oliver.

**Pull request merged**